packages/web-app-admin-settings/src/views/Users.vue

```diff
@@ -500,70 +502,157 @@ export default defineComponent({
     }
 
     const addUsersToGroups = async ({ users: affectedUsers, groups: groupsToAdd }) => {
-      try {
-        const client = clientService.graphAuthenticated
-        const usersToFetch = []
-        const addUsersToGroupsRequests = []
-        groupsToAdd.reduce((acc, group) => {
-          for (const user of affectedUsers) {
-            if (!user.memberOf.find((userGroup) => userGroup.id === group.id)) {
-              acc.push(client.groups.addMember(group.id, user.id, configurationManager.serverUrl))
-              if (!usersToFetch.includes(user.id)) {
-                usersToFetch.push(user.id)
-              }
+      const client = clientService.graphAuthenticated
+      const usersToFetch = []
+      const promises = groupsToAdd.reduce((acc, group) => {
+        for (const user of affectedUsers) {
+          if (!user.memberOf.find((userGroup) => userGroup.id === group.id)) {
+            acc.push(client.groups.addMember(group.id, user.id, configurationManager.serverUrl))
+            if (!usersToFetch.includes(user.id)) {
+              usersToFetch.push(user.id)
             }
           }
-          return acc
-        }, addUsersToGroupsRequests)
-        const usersResponse = await loadingService.addTask(async () => {
-          await Promise.all(addUsersToGroupsRequests)
-          return Promise.all(usersToFetch.map((userId) => client.users.getUser(userId)))
-        })
-        updateLocalUsers(usersResponse.map((r) => r.data))
-        await store.dispatch('showMessage', {
-          title: $gettext('Users were added to groups successfully')
-        })
+        }
+        return acc
+      }, [])
+      if (!promises.length) {
+        const title = $ngettext(
+          'Group assignment already added',
+          'Group assignments already added',
+          affectedUsers.length * groupsToAdd.length
+        )
+        store.dispatch('showMessage', { title })
         addToGroupsModalIsOpen.value = false
+        return
+      }
+
+      const results = await loadingService.addTask(() => {
+        return Promise.allSettled(promises)
+      })
+
+      const succeeded = results.filter((r) => r.status === 'fulfilled')
+      if (succeeded.length) {
+        const title =
+          succeeded.length === 1 && groupsToAdd.length === 1 && affectedUsers.length === 1
+            ? $gettext('Group assignment "%{group}" was added successfully', {
+                group: groupsToAdd[0].displayName
+              })
+            : $ngettext(
+                '%{groupAssignmentCount} group assignment was added successfully',
+                '%{groupAssignmentCount} group assignments were added successfully',
```

Comment:
I find the term "group assignment" to be quite technical. The action is labeled "Add to groups", I'd prefer to stay in that terminology, so e.g.  "Users have been added to the selected groups". Ultimately @tbsbdr decides of course.

Reply:
Problem is the success/error count if we want group assignment consist of n groups * n users if we want to keep the n users had been added to group, n users failed added to group we need to analyse the requests and add extra logic

Reply:
Okay I understand. What about simply "n groups were added successfully"?

Reply:
So with n is users * groups ? That's fine for me

Reply:
Arghs no, now I fully understand, sorry! That might be confusing as well for the user. Hm. Then I don't have a better idea than "group assignment". That would be "Gruppenzugehörigkeit" in German? 😂

Reply:
Or ze very well "Gruppenzuweisung"